### ISSUE TITLE
False "match must be exhaustive" when combined with Fn

### ISSUE THREAD

**Issue by qiujiangkun** [intellij-rust/intellij-rust#6137]
There is no compile error but Intellij Rust gives a false error.
![image](https://user-images.githubusercontent.com/33482468/93667827-44e66680-fabb-11ea-8c28-83020b315c45.png)
```rust
enum NameIteration {
    KeepGoing,
    Stop(Result<(), std::io::Error>),
}
fn test(f: &dyn Fn(&str) -> NameIteration) -> Result<(), std::io::Error> {
    match f("name") {
        NameIteration::Stop(result) => {
            return result;
        },
        NameIteration::KeepGoing => {},
    }
    Ok(())
}
```

**Comment by Undin** [comment]
Looks like it was fixed in #5720
